tests/jobs/ahm_analysis/test_ahm_analysis.py

```diff
@@ -0,0 +1,418 @@
+import os
+
+import pytest
+import pandas as pd
+import numpy as np
+
+import semeio.workflows.ahm_analysis.ahmanalysis as ahmanalysis
+from semeio._exceptions.exceptions import ValidationError
+
+from unittest.mock import MagicMock
+from itertools import product
+from pathlib import Path
+
+from scipy import stats
+from ecl.grid import EclGridGenerator
+
+
+@pytest.mark.usefixtures("setup_tmpdir")
+def test_make_update_log_df(test_data_root):
+    """test function creates a dataframe from update_log file
+    and replace '...' with key_obs"""
+    test_data_dir = os.path.join(test_data_root, "update_log/oneobs")
```

Comment:
I mentioned that the input test case was a bit trivial, and for instance the `allobs` updatelog would yield an unexpected result. If one is to replace with `allobs` test case, this test wouldn't fail on the unexpected issue, it would merely fail on the status not being `Active` at the same indexes (a happy coincident perhaps).

My point is, let's be more exhaustive of what we want to test, let's assert the output instead (or a larger part of it).

We could for instance assert the part we're mostly interested in, where there have been changes to the `...` value:

```
expected = pd.DataFrame(
        columns=["obs_key", "obs_mean"],
        data=[
            ("OP_1_WGOR2", 170.00),
            ("RWI_3_OBS", 285.00),
            ("RWI_3_OBS", 258.00),
            ("RWI_3_OBS", 271.00),
            ("RWI_2_OBS", 291.00),
            ("RWI_2_OBS", 233.00),
            ("RWI_2_OBS", 297.00),
            ("ROP_5_OBS", 285.00),
            ("ROP_5_OBS", 279.00),
            ("ROP_5_OBS", 288.00),
            ("ROP_4_OBS", 292.00),
            ("ROP_4_OBS", 268.00),
            ("ROP_4_OBS", 280.00),
        ],
    )
```

Reply:
done, allobs updatelog is used and check is made that lines with ... in input data are replaced by names of the observation.